common/CommonAPI.h

```diff
@@ -722,23 +711,281 @@ class CommonAPI
 };
 
 #ifndef _NBL_PLATFORM_ANDROID_
-class GraphicalApplication : public nbl::system::IApplicationFramework, public nbl::ui::IGraphicalApplicationFramework
+class GraphicalApplication : public CommonAPI::CommonAPIEventCallback, public nbl::system::IApplicationFramework, public nbl::ui::IGraphicalApplicationFramework
 {
 protected:
 	~GraphicalApplication() {}
+
+	static constexpr uint64_t MAX_TIMEOUT = 99999999999999ull;
+
+	uint32_t m_frameIx = 0;
+	nbl::core::deque<CommonAPI::IRetiredSwapchainResources*> m_qRetiredSwapchainResources;
+	uint32_t m_swapchainIteration = 0;
+	std::array<uint32_t, CommonAPI::InitOutput::MaxSwapChainImageCount> m_imageSwapchainIterations;
+	std::mutex m_swapchainPtrMutex;
+
+	// Returns retired resources
+	virtual std::unique_ptr<CommonAPI::IRetiredSwapchainResources> onCreateResourcesWithSwapchain(const uint32_t imageIndex) { return nullptr; }
```

Comment:
this should probably take a `const CResizableSwapchainUtils* swapchain` in case we make an example with multiple resizable swapchains in the future (to tell apart which swapchain we want to deal with)

Reply:
also to automatically set `retiredResources->retiredFrameId = swapchainUtils->m_frameIx;` on the returned/created item

(then we make this an `inline` and make a `virtual unique_ptr<> onCreateResourcesWithSwapchain_impl`)